consensus/src/pipeline/virtual_processor/utxo_validation.rs

```diff
@@ -138,7 +139,10 @@ impl VirtualStateProcessor {
 
         // Make sure accepted tx ids are sorted before building the merkle root
         // NOTE: when subnetworks will be enabled, the sort should consider them in order to allow grouping under a merkle subtree
-        ctx.accepted_tx_ids.sort();
+        // Preserve canonical order of accepted transactions after hard-fork
+        if !self.accepted_id_merkle_root.is_active(pov_daa_score) {
+            ctx.accepted_tx_ids.sort();
```

Comment:
May we `sort_unstable()` instead?
```suggestion
            ctx.accepted_tx_ids.sort_unstable();
```

It's faster than regular sort, though it won't preserve order of equal elements.

Reply:
It was originally sorted this way, so I should preserve past behaviour, otherwise network will split ....